### PR TITLE
Regenerate browserslist config and drop iOS 10 and Safari 9 support

### PR DESCRIPTION
**Pull request by taneliang**
Our browserslist config was last updated ~1.5 years ago in #1697. This PR regenerates the config and performs other chores in the codebase.

TL;DR:

1. cd packages/browserslist-config-nusmods
1. yarn start
2. cd ../../website
3. yarn upgrade (mainly to bump browserslist to the latest version, but also just because)

### Test plan

- Manually tested on an iOS 11 device on Browserstack.
- `yarn e2e`:
	- `safariMobile` env fails when run as part of `yarn e2e`: 
		image

	- But `yarn e2e -e safariMobile` works:
		image

### Unrelated side notes

- @tippyjs/react is now @tippy.js/react. I'll upgrade this in a future PR after this is merged.
- We can't upgrade to Node 14 until webpack-dev-server is upgraded to v4. v4 is currently in beta.
- eslint-loader has been deprecated and replaced by eslint-webpack-plugin. I'll upgrade this in a future PR after this is merged.

### CODE REVIEW

**Review by li-kai — APPROVED** [review]
LGTM 🥦
